docs/user/ppl/admin/settings.rst

```diff
@@ -73,6 +73,42 @@ PPL query::
       "status": 400
     }
 
+plugins.ppl.query.timeout
+=========================
+
+Description
+-----------
+
+This setting controls the maximum execution time for PPL queries. When a query exceeds this timeout, it will be interrupted and return a timeout error.
+
+1. The default value is 120s (2 minutes).
```

Comment:
Haven't go to details, curious on the default value.
1. why 120s, can we increase this default value? The default value of `join.subsearch_timeout` in SPL is 120s, maybe this could be larger.
2. can we disable this timeout by set it to a negative value, (similar to `max_time_per_process` in SPL.

Reply:
Updated the default to 5 minutes

From measuring production clusters the highest timeout I saw was 30 seconds, figured 120 was a decent default threshold since it can always be increased